test/test/custom_activation.cpp

```diff
@@ -50,3 +50,45 @@ TEST_CASE("custom_activation")
     winrt_activation_handler = nullptr;
     clear_factory_cache();
 }
+
+// An activation handler that can only deal with Windows.Foundation.Uri.
+bool invoked_unimplemented = false;
+bool __stdcall uri_only_activation_handler(void* classId, winrt::guid const& iid, void** factory, int32_t* hr) noexcept
+{
+    using namespace std::literals::string_view_literals;
+
+    std::wstring_view const name{ *reinterpret_cast<winrt::hstring*>(&classId) };
+    // custom_factory is only interested in Windows.Foundation.Uri
+    if (name == L"Windows.Foundation.Uri"sv) {
+        *factory = detach_abi(make<custom_factory>());
```

Comment:
This needs to honor the `iid`.

```suggestion
        *hr = make<custom_factory>().as(iid, factory);
```

Reply:
When I saw your first comment I considered doing the fix in another PR, but since you mentioned here I took the freedom to do some more related refactoring on the existing test case as well.